src/hotspot/share/nmt/memReporter.cpp

```diff
@@ -101,6 +101,9 @@ void MemReporterBase::print_malloc(const MemoryCounter* c, MemTag mem_tag) const
     out->print(" (peak=%zu%s #%zu)",
         amount_in_current_scale(pk_amount), scale, pk_count);
   }
+  size_t overhead = c->allocated() - c->requested();
```

Comment:
Maybe better to have an `assert(c->allocated() >= c->requested(), "should be");`.